### PR TITLE
Renamed folder and namespace network2 to http.

### PR DESCRIPTION
**Pull request by andescu**
To avoid multiple commits that will rename network2 to network after
porting is completed it has been decided to rename namespace and folder
to _http_. This is a more descriptive name for the actual use of the
Network classes as the main point of http(s) requests handler.

Relates-to: OLPEDGE-368

Signed-off-by: Andrei Popescu <andrei.popescu@here.com>

### CODE REVIEW

File: olp-cpp-sdk-core/CMakeLists.txt [hunk 1]
```diff
@@ -153,19 +151,19 @@ file(GLOB NETWORK_ANDROID_SOURCE
     "${CMAKE_CURRENT_SOURCE_DIR}/src/network/socket/*.cpp"
 )
     set(NETWORK_SOURCE ${NETWORK_SOURCE} ${NETWORK_ANDROID_SOURCE})
-    set(NETWORK2_SOURCE ${NETWORK2_SOURCE} ${NETWORK2_ANDROID_SOURCE})
+    set(EDGE_SDK_HTTP_SOURCES ${EDGE_SDK_HTTP_SOURCES} ${http_ANDROID_SOURCE})
```
Comment (serhiikobalt) [inline olp-cpp-sdk-core/CMakeLists.txt]: ```HTTP_ANDROID_SOURCE``` . I mean going all caps.

File: olp-cpp-sdk-core/CMakeLists.txt [hunk 2]
```diff
@@ -138,12 +138,10 @@ file(GLOB NETWORK_SOURCE
...
     "${CMAKE_CURRENT_SOURCE_DIR}/src/network/*.h"
 )
-set(NETWORK2_SOURCE
-    "${CMAKE_CURRENT_SOURCE_DIR}/src/network2/NetworkProxySettings.cpp"
-    "${CMAKE_CURRENT_SOURCE_DIR}/src/network2/NetworkRequest.cpp"
-    "${CMAKE_CURRENT_SOURCE_DIR}/src/network2/NetworkResponse.cpp"
-    "${CMAKE_CURRENT_SOURCE_DIR}/src/network2/NetworkSettings.cpp"
-    "${CMAKE_CURRENT_SOURCE_DIR}/src/network2/Network.cpp"
+
+file(GLOB EDGE_SDK_HTTP_SOURCES
```
Comment (serhiikobalt) [inline olp-cpp-sdk-core/CMakeLists.txt]: Why would you switch to GLOBing sources here, but then not for EDGE_SDK_HTTP_HEADERS?

File: olp-cpp-sdk-core/CMakeLists.txt [hunk 3]
```diff
@@ -153,19 +151,19 @@ file(GLOB NETWORK_ANDROID_SOURCE
...
 endif()
 if (IOS)
     set(NETWORK_SOURCE ${NETWORK_SOURCE} ${NETWORK_IOS_SOURCES})
-    set(NETWORK2_SOURCE ${NETWORK2_SOURCE} ${NETWORK2_IOS_SOURCES})
+    set(EDGE_SDK_HTTP_SOURCES ${EDGE_SDK_HTTP_SOURCES} ${http_IOS_SOURCES})
 endif()
 if (WIN32)
     set(NETWORK_SOURCE ${NETWORK_SOURCE} ${NETWORK_WINHTTP_SOURCES})
-    set(NETWORK2_SOURCE ${NETWORK2_SOURCE} ${NETWORK2_WINHTTP_SOURCES})
+    set(EDGE_SDK_HTTP_SOURCES ${EDGE_SDK_HTTP_SOURCES} ${NETWORK_WINHTTP_SOURCES})
```
Comment (serhiikobalt) [inline olp-cpp-sdk-core/CMakeLists.txt]: ```NETWORK_WINHTTP_SOURCES``` is taken by existing winhttp network implementation

File: olp-cpp-sdk-core/CMakeLists.txt (hunk 1 above)
Comment (andescu) [reply olp-cpp-sdk-core/CMakeLists.txt]: Find/replace error.

File: olp-cpp-sdk-core/CMakeLists.txt [hunk 4]
```diff
@@ -138,34 +138,34 @@ file(GLOB NETWORK_SOURCE
...
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/http/NetworkResponse.cpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/http/NetworkSettings.cpp"
 )
 
 if (ANDROID)
 file(GLOB NETWORK_ANDROID_SOURCE
     "${CMAKE_CURRENT_SOURCE_DIR}/src/network/android/*.cpp"
     "${CMAKE_CURRENT_SOURCE_DIR}/src/network/android/*.h"
-    "${CMAKE_CURRENT_SOURCE_DIR}/src/network/socket/*.cpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/netwofrk/socket/*.cpp"
```
Comment (serhiikobalt) [inline olp-cpp-sdk-core/CMakeLists.txt]: oops

Comment (andescu) [reply olp-cpp-sdk-core/CMakeLists.txt]: :) New rule -> Never push code before you finish your first coffee.

**Pull request merged**